src/internal/packager/helm/post-render.go

```diff
@@ -117,29 +118,35 @@ func (r *renderer) Run(renderedManifests *bytes.Buffer) (*bytes.Buffer, error) {
 
 func (r *renderer) adoptAndUpdateNamespaces(ctx context.Context) error {
 	c := r.cluster
-	existingNamespaces, _ := c.GetNamespaces(ctx)
+	namespaceList, err := r.cluster.Clientset.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
 	for name, namespace := range r.namespaces {
 
 		// Check to see if this namespace already exists
 		var existingNamespace bool
-		for _, serverNamespace := range existingNamespaces.Items {
+		for _, serverNamespace := range namespaceList.Items {
 			if serverNamespace.Name == name {
 				existingNamespace = true
 			}
 		}
 
 		if !existingNamespace {
 			// This is a new namespace, add it
-			if _, err := c.CreateNamespace(ctx, namespace); err != nil {
+			_, err := c.Clientset.CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+			if err != nil {
 				return fmt.Errorf("unable to create the missing namespace %s", name)
 			}
 		} else if r.cfg.DeployOpts.AdoptExistingResources {
-			if r.cluster.IsInitialNamespace(name) {
+			// IsInitialNamespace returns true if the given namespace name is an initial k8s namespace: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#initial-namespaces
```

Comment:
Can we update this comment? The `IsInitialNamespace` function is deleted in this PR

Reply:
I also realized that the logic was wrong so I fixed it. Checking for the prefix would give false positives for namespaces that begin with `kube-`.

Reply:
Yeah I think it's fine to be more precise with the default namespace checks. The k8s docs recommend to avoid creating namespaces with `kube-` prefix, but we don't necessarily have to enforce that in our logic.

https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#working-with-namespaces

> Avoid creating namespaces with the prefix `kube-`, since it is reserved for Kubernetes system namespaces.